### DISCUSSION TITLE
Module not found: Can't resolve 'v8'

### DISCUSSION THREAD

**Discussion by zhangwei900808** [nextauthjs/next-auth#5068] [Help]
### Environment

  System:
    OS: macOS 12.5
    CPU: (8) x64 Intel(R) Core(TM) i7-4770HQ CPU @ 2.20GHz
    Memory: 504.44 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 14.17.1 - /usr/local/bin/node
    Yarn: 1.22.5 - ~/.yarn/bin/yarn
    npm: 8.10.0 - /usr/local/bin/npm
  Browsers:
    Chrome: 103.0.5060.134
    Edge: 94.0.992.50
    Firefox: 102.0.1
    Safari: 15.6
  npmPackages:
    next: ^12.2.0 => 12.2.2 
    next-auth: ^4.10.3 => 4.10.3 
    react: ^18.1.0 => 18.2.0 

### Reproduction URL

...

### Describe the issue

**pages/index.js**
```
import { authOptions } from '@/pages/api/auth/[...nextauth]'
import { unstable_getServerSession } from "next-auth/next"


export const getServerSideProps = wrapper.getServerSideProps(store => async (ctx) => {
  const session = await unstable_getServerSession(ctx.req, ctx.res, authOptions)
  if(session){
    await store.dispatch(setAuth({
      ...session
    }))
  }
  return {
    props: {
      
    }
  };
});
```

## when I config next-auth and run it but its not work!
```
next-dev.js?3515:24 ./node_modules/openid-client/lib/helpers/keystore.js:1:0
Module not found: Can't resolve 'v8'

Import trace for requested module:
./node_modules/openid-client/lib/client.js
./node_modules/openid-client/lib/issuer.js
./node_modules/openid-client/lib/index.js
./node_modules/next-auth/core/lib/oauth/callback.js
./node_modules/next-auth/core/routes/callback.js
./node_modules/next-auth/core/routes/index.js
./node_modules/next-auth/core/index.js
./node_modules/next-auth/next/index.js
./src/hoc/useAuthProps.js
./src/pages/index.js

https://nextjs.org/docs/messages/module-not-found
```

### How to reproduce

1、run in Chrome
2、error

### Expected behavior

its worked!

**Comment by zhangwei900808** [comment]
I resolve it 
**but when I used `unstable_getServerSession` in new single file it not Work!!!** 
```
import {wrapper} from '@/store'
import {unstable_getServerSession} from "next-auth/next";
import { authOptions } from '@/pages/api/auth/[...nextauth]'
import {setAuth} from '@/store/slices/authSlice'

export function useAuthProps(callback){
  return wrapper.getServerSideProps((store) =>
    async ({ req, res }) => {
      const session = await unstable_getServerSession(req, res, authOptions)
      return callback(store, req, res)
    })
}
```

**Comment by faeb187** [comment]
You can't run server-side code in a browser.

Explanation: NodeJS runs in V8 (written in C++)
NodeJS can't start because not found on a client.

Here an example how I've implemented it using the appDir (NextJS 13 feature).
It's actually the opposite, as the server would throw after trying to resolve `useEffect()`

`"use client"` on top of a file excludes the server from running it:

 Example: /app/home/page.tsx

```js
"use client";

import i18n from "app/i18n";
import Head from "next/head";
import { usePathname } from "next/navigation";
import { useEffect, useState } from "react";
import { App } from "server/models/AppModel";

/**
 * Home
 * @tags #home #page
 */
export default function () {
  const [app, setApp] = useState<App | null>(null);
  const path = usePathname()?.slice(1);

  useEffect(() => {
    try {
      fetch("/api/home").then(async (response: Response) => {
        setApp((await response.json()) as App | null);
      });
      /*fetch("/api/home", { method: "POST" }).then((value) => {
        console.log(value);
      });*/
    } catch (error: any) {
      console.error(error);
    }
  }, []);

  return (
    <>
      <Head>
        <title>{`AnkhHome`}</title>
        <meta name="description" content={`Ankhorage /home`} />
        <link rel="icon" href="/favicon.ico" />
      </Head>
      <>
        <h1>{path}</h1>
        {app && (
          <p>
            {i18n.t("app")}: {app.title}
          </p>
        )}
      </>
    </>
  );
}

```

****
Like in e.g. MeteorJS we're living in a isomorphic environment now, where **code is meant to run on client and server** (same file).
This comes with great benefits, *coughs* ... first things that come to mind:

- same form validator running on client & server, updates can be done once in one file, half costs, half tests, half everything. Only joy multiplies.
- therefore no misunderstandings and out-of-sync code between backend and frontend team anymore
- no knowledge separation, everyone is full stack, knowledge-base will be smaller but deeper (nobody said you can't work with other languages for other purposes (and using same lang client/server sounds like a charm doesn't it?)
- therefore the whole team can communicate on a different level, you won't need 4 experts in 4 languages and build stitches and bridges and adapters and so on in-between (less complexity, more transparency)
- it's something companies lose a lot of money without even know it's gone (could talk hours about it)

**Comment by ax0n-pr1me** [comment]
For anyone stumbling on this as an upgrade issue... I was able to resolve it by changing the 

`import { UserProvider } from '@auth0/nextjs-auth0';`. 

to 

`import { UserProvider } from '@auth0/nextjs-auth0/client';`

in `_app.js`
